xarray/core/datatree.py

```diff
@@ -1279,12 +1267,12 @@ def equals(self, other: DataTree) -> bool:
         DataTree.isomorphic
         DataTree.identical
         """
-        if not self.isomorphic(other, strict_names=True):
+        if not self.isomorphic(other):
             return False
 
         return all(
             node.dataset.equals(other_node.dataset)
```

Comment:
A little clearer intent?
```suggestion
            # do not check that inherited coordinates are defined on the same level
            node.to_dataset(inherit=True).equals(other_node.to_dataset(inherit=True))
```